generator/integration_tests/tests/golden_kitchen_sink_stub_factory_test.cc

```diff
@@ -41,7 +41,9 @@ class GoldenKitchenSinkStubFactoryTest : public ::testing::Test {
 };
 
 TEST_F(GoldenKitchenSinkStubFactoryTest, DefaultStubWithoutLogging) {
-  auto default_stub = CreateDefaultGoldenKitchenSinkStub(CompletionQueue{}, {});
+  Options options;
```

Comment:
can we set expectations on `options.get<EndpointOption>()` on the other side of the `CreateDefault*Stub` call?

I know https://github.com/scotthart/google-cloud-cpp/blob/main/generator/integration_tests/tests/universe_domain_test.cc is a thing... just seems like we should have a unit test here, too.